modules/rtm/src/RTM/twostream/two.stream.f90

```diff
@@ -0,0 +1,90 @@
+subroutine two_stream(rl, tl, solar_zenith, L, soil_moisture, &     ! Inputs
+        alpha_c, alpha_i, Tc, Ti, Ac, Ai)       ! Outputs
+
+    use mod_dataspec_wavelength
+    use mod_hapke_soil
+    use mod_biggamma
+
+    implicit none
+    real*8, intent(in), dimension(nw) :: rl, tl
+    real*8, intent(in) :: L, solar_zenith, soil_moisture
+
+    real*8, intent(out), dimension(nw) :: alpha_c, alpha_i, Tc, Ti, Ac, Ai
+
+    !! TODO: More intelligent treatment of zeta and zetastar
+    !! TODO: Scale leaf optics by forward/backward scattering efficiency (depends on leaf orientation function)
+    !! TODO: Implement alternative G functions (as callable functions of mu)
+
+    !! Fixed parameters
+    real*8, parameter :: mui = 0.5d0 / 0.705d0      ! Isotropic cosine constant
+    real*8, parameter :: G = 0.5d0                ! Structure factor for spherically (randomly) distributed leaf angles
+    real*8, parameter :: zeta = 0.67d0          ! Structure factor for collimated radiation (see Pinty 2006)
+    real*8, parameter :: zetastar_factor = 1.0d0    ! Structure factor scaling term for isotropic radiation (see Pinty 2006)
+
+    !! Derived quantities
+    real*8 :: mu0, KL, zetastar
+    real*8 :: tau_c, tau_i, exptaui             ! Optical depths for collimated (c) and isotropic (i) radiation
+    real*8, dimension(nw, 4) :: gammas_c, gammas_i      ! gamma terms for Meador & Weaver (1980) solutions
+    real*8, dimension(nw) :: w0, rho_soil               ! Canopy albedo, soil reflectance
+    real*8, dimension(nw) :: RcBB, TcBB, RiBB, TiBB     ! Black background reflectance and transmittance
+    real*8 :: TcBC, TiBC                                ! Black canopy transmittances
+    real*8, dimension(nw) :: Rstar                      ! Infinite series reflectance term
+    real*8, dimension(nw) :: TcT, TiT                   ! Total one-way transmittances
+    real*8, dimension(nw) :: rhoxTiT
```

Comment:
It's an intermediate quantity that saves a few calculations. I only bothered defining terms that are physically meaningful, and that map onto terms in the reference literature.